fix: enable keyboard navigation in the picker

musale opened the pull request:
<!-- Review contributing guidelines before creating PRs -->
<!-- https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/main/CONTRIBUTING.md -->

Closes #2305  <!-- REQUIRED: Add the issue number (ex: #12) so the issue is automatically closed when PR is merged -->

### PR Type
<!-- Please uncomment one or more that apply to this PR -->

- Bugfix
<!-- - Feature -->
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Other... Please describe: -->

### Description of the changes
`mgt-picker` uses a combobox that does not have support for keyboard events on the options. This fix uses the combobox and options values to trigger the correct action when you select an option.

### PR checklist
- [x] Project builds (`yarn build`) and changes have been tested in at least two supported browsers (Edge + non-Chromium based browser)
- [x] All public APIs (classes, methods, etc) have been documented following the jsdoc syntax
- [x] [Stories](https://github.com/microsoftgraph/microsoft-graph-toolkit/wiki/Storybook#writing-stories) have been added and existing stories have been tested
- [ ] Added appropriate [documentation](https://github.com/microsoftgraph/microsoft-graph-toolkit/wiki/Documentation). Docs PR: <!-- Link to docs PR here -->
<!-- Please uncomment if any Accessibility related changes -->
- [x] Accessibility tested and approved-->
- [ ] License header has been added to all new source files (`yarn setLicense`)
- [ ] Contains **NO** breaking changes

### Other information
<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. 
     Please note that breaking changes are likely to be rejected -->

musale commented on packages/mgt-components/src/components/mgt-picker/mgt-picker.ts:
```diff
@@ -269,4 +277,28 @@ export class MgtPicker extends MgtTemplatedComponent {
+      value = fluentOptionEl.getAttribute('value');
+    }
+
+    if (['Enter', 'Backspace'].includes(keyName)) {
```
This component has a very _weird_ build. In my keyboard, `e.key` shows up as `Backspace`. Now, I have to change that to Tab key because the space bar edits the combobox and overriding it for any other functionality renders searches with a space undoable. The Tab can select an option so I have added it in to fire the custom event.

gavinbarron replied:
Come to think of it shouldn't space key hander be on the fluent-option? 
That way the input doesn't bubble up to the input

musale replied:
Yes, it should. However, the fluent-option has no events it's emitting when I put a `keydown` and `keyup` handlers to it. Being a combobox, the text area and the menu are linked. Maybe to allow typing searches? That's why I implemented the events on the fluent-menu and then handle the selected fluent-option by reading in the DOM children it has and filtering the required option.

sebastienlevert commented:
@vagpt and @KarunaNarang this item is ready for review.

vagpt commented:
Hi @gavinbarron,

Hope you are doing well.

As per the above comment **'This branch is out of date with the base branch.'**, Can we test this issue or should we wait for updated storybook URL? 
As of now this issue is still repro.

CC: @KarunaNarang

vagpt commented:
Hi @gavinbarron,

Hope you are doing well.

As per the above comment **'This branch is out of date with the base branch.'**, Can we test this issue or should we wait for updated storybook URL? 
As of now this iisue is still repro.

CC: @KarunaNarang

musale commented:
> Hi @gavinbarron,
> 
> Hope you are doing well.
> 
> As per the above comment **'This branch is out of date with the base branch.'**, Can we test this issue or should we wait for updated storybook URL? As of now this issue is still repro.
> 
> CC: @KarunaNarang

You can always test despite that comment because the current storybook will have the relevant parts that we want to get tested. I have updated the changes to fix a merge issue and updated the base branch. You can now retest.

vagpt commented:
> > Hi @gavinbarron,
> > Hope you are doing well.
> > As per the above comment **'This branch is out of date with the base branch.'**, Can we test this issue or should we wait for updated storybook URL? As of now this issue is still repro.
> > CC: @KarunaNarang
> 
> You can always test despite that comment because the current storybook will have the relevant parts that we want to get tested. I have updated the changes to fix a merge issue and updated the base branch. You can now retest.

Hi @musale,

Thanks for the quick reply.

I have validate this issue using the provided URL by you and this issue is partially fixed on the environment below. As of now after expanding the combo box, a code part is getting displayed with the Task list item.

**Test Environment:**
URL: https://mgt.dev/next/pr/2324/?path=/story/samples-general--localization
OS Build: Windows 11
Version: 22H2 (OS Build 25375.1)
Browser: Edge dev
Browser Version 115.0.1880.1 (Official build) dev (64-bit)

**Snippet:**
![image](https://github.com/microsoftgraph/microsoft-graph-toolkit/assets/95913748/54c4071e-8a18-4624-acf0-9c918dcf4a30)

vagpt commented:
> > > > Hi @gavinbarron,
> > > > Hope you are doing well.
> > > > As per the above comment **'This branch is out of date with the base branch.'**, Can we test this issue or should we wait for updated storybook URL? As of now this issue is still repro.
> > > > CC: @KarunaNarang
> > > 
> > > 
> > > You can always test despite that comment because the current storybook will have the relevant parts that we want to get tested. I have updated the changes to fix a merge issue and updated the base branch. You can now retest.
> > 
> > 
> > Hi @musale,
> > Thanks for the quick reply.
> > I have validate this issue using the provided URL by you and this issue is partially fixed on the environment below. As of now after expanding the combo box, a code part is getting displayed with the Task list item.
> > **Test Environment:** URL: https://mgt.dev/next/pr/2324/?path=/story/samples-general--localization OS Build: Windows 11 Version: 22H2 (OS Build 25375.1) Browser: Edge dev Browser Version 115.0.1880.1 (Official build) dev (64-bit)
> > **Snippet:** ![image](https://user-images.githubusercontent.com/95913748/243318115-54c4071e-8a18-4624-acf0-9c918dcf4a30.png)
> 
> Hmm, that's strange because it's what I have fixed. Could you hard reload and retry?

Thanks @musale 

Now it is working as expected.
